base/audit/auditd_types.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package audit
+
+// auditdModuleDescriptor describes an audit module descriptor in the auditd JSON format.
+type auditdModuleDescriptor struct {
+	Version uint                    `json:"version"`
+	Module  string                  `json:"module"`
+	Events  []auditdEventDescriptor `json:"events"`
+}
+
+// auditdEventDescriptor describes an audit event in the auditd JSON format.
+type auditdEventDescriptor struct {
+	ID                 ID             `json:"id"`
+	Name               string         `json:"name"`
+	Description        string         `json:"description"`
+	Sync               bool           `json:"sync,omitempty"`
+	Enabled            bool           `json:"enabled,omitempty"`
+	FilteringPermitted bool           `json:"filtering_permitted,omitempty"`
+	MandatoryFields    map[string]any `json:"mandatory_fields"`
+	OptionalFields     map[string]any `json:"optional_fields,omitempty"`
+}
+
+// toAuditdEventDescriptor converts an eventDescriptor to an auditdEventDescriptor.
+// These are _mostly_ the same, but each event holds its own ID in an array in the JSON format.
+func toAuditdEventDescriptor(id ID, e eventDescriptor) auditdEventDescriptor {
+	return auditdEventDescriptor{
+		ID:                 id,
+		Name:               e.name,
+		Description:        e.description,
+		Enabled:            e.enabledByDefault,
+		FilteringPermitted: e.filteringPermitted,
+		MandatoryFields:    toAuditdFieldType(e.mandatoryFields),
+		OptionalFields:     toAuditdFieldType(e.optionalFields),
+	}
+}
+
+// auditd values that represent each type of field
+const (
+	auditFieldTypeNumber = 1
+	auditFieldTypeString = ""
+	auditFieldTypeBool   = true
```

Comment:
I think I'd rename these types because it was not obvious to me why some were default golang values and some are not.

This matches the spec for couchbase server and I wonder if it's worth linking this spec.

Reply:
What should I rename them to? I figured auditFieldType was enough?

I'll definitely link the spec because it's weird.

Reply:
I guess I was thinking `auditFieldBoolDefault`, etc